src/ResourceManager/Common/Commands.TestFx/TestClientFactory.cs

```diff
@@ -0,0 +1,144 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Graph.RBAC.Version1_6;
+using Microsoft.Rest;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+
+namespace  Microsoft.Azure.Commands.TestFx
+{
+    public class TestClientFactory : IClientFactory
+    {
+        private readonly MockContext _mockContext;
+
+        public TestClientFactory(MockContext mockContext)
+        {
+            if (mockContext == null) throw new ArgumentNullException(nameof(mockContext));
+            _mockContext = mockContext;
+        }
+
+        public TClient CreateArmClient<TClient>(IAzureContext context, string endpoint) where TClient : ServiceClient<TClient>
+        {
+            if (typeof(TClient) != typeof(GraphRbacManagementClient))
+            {
+                return _mockContext.GetServiceClient<TClient>();
+            }
+
+            var graphClient = _mockContext.GetGraphServiceClient<GraphRbacManagementClient>();
+            graphClient.TenantID = context.Tenant.Id;
```

Comment:
Does [`_mockContext.GetGraphServiceClient<GraphRbacManagementClient>();`](https://github.com/Azure/azure-powershell/pull/6651/files#diff-69c48711471abbd087a8100d64adbee7R48) not return a `graphClient` with the `TenaniId` initiad?

Reply:
When I debugged this - the ```TenantId``` was not initialized, that's why I added the line #48.

Reply:
Is it the same behavior in Playback, REcord and Live modes?
Might be better to  just add a check to see if the `graphClient.TenantID` is non null.. in case its null only then set it to the tenentId in the context...

Reply:
Context initialization [happens](https://github.com/vladimir-shcherbakov/azure-powershell/blob/iss%236180/src/ResourceManager/Common/Commands.TestFx/TestManager.cs#L268) in TestManager and depends on a test mode. I doubt mock context does any client initialization.